engine/src/main/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputUtils.java

```diff
@@ -288,6 +288,48 @@ public static final String getLine( LogChannelInterface log, InputStreamReader r
     return getLine( log, reader, type, formatNr, line );
   }
 
+  public static final String getLine( LogChannelInterface log, InputStreamReader reader, EncodingType encodingType,
+                                      int fileFormatType, StringBuilder line, String regex )
+    throws KettleFileException {
+
+    return getLine( log, reader, encodingType, fileFormatType, line, regex, 0 )[0];
+
+  }
+
+  /**
+   *
+   * Returns in the first position a line; ;
+   * on the second position how many lines from file were read to get a full line
+   *
+   */
+  public static final String[] getLine( LogChannelInterface log, InputStreamReader reader, EncodingType encodingType,
+                                      int fileFormatType, StringBuilder line, String regex, long lineNumberInFile )
+    throws KettleFileException {
+
+    String sline = getLine( log, reader, encodingType, fileFormatType, line );
+
+    while ( sline != null ) {
+    /*
+    Check that the number of enclosures in a line is even.
+    If not even it means that there was an enclosed line break.
+    We need to read the next line(s) to get the remaining data in this row.
+    */
+      if ( checkPattern( sline, regex ) % 2 == 0 ) {
+        return new String[] { sline, String.valueOf( lineNumberInFile ) };
+      }
+
+      String nextLine = getLine( log, reader, encodingType, fileFormatType, line );
+
+      if ( nextLine == null ) {
+        break;
+      }
+
+      sline = sline + nextLine;
+      lineNumberInFile++;
+    }
+    return new String[] { sline, String.valueOf( lineNumberInFile ) };
```

Comment:
This seems like a hack to me. Instead of an array, shouldn't you use a simple class with the string and the line number in the file? Would be much cleaner looking, and avoid data conversion. For a 10-line file, the String.valueOf won't be a problem. In a billion-line file though, it will consume a significant amount of time.